### PR TITLE
build-tracker: deploy to MSP

### PR DESCRIPTION
**Pull request by Strum355**
Bring it into [year 2024] :tada: will be building on this as part of #60455 OKR, so having a more convenient deploy method would be cool

MSP PR: https://github.com/sourcegraph/managed-services/pull/1011

## Test plan

testing this live :eye:

### CODE REVIEW

File: dev/build-tracker/BUILD.bazel [hunk 1]
```diff
@@ -36,9 +47,58 @@ go_test(
...
+    tags = [
+        "exclusive",
+        "requires-network",
+    ],
+)
+
+oci_push(
+    name = "candidate_push",
+    image = ":image",
+    repository = image_repository("msp-example"),
```
Comment (jac) [inline dev/build-tracker/BUILD.bazel:96]:
```suggestion
    repository = image_repository("build-tracker"),
```

File: dev/build-tracker/integration_test.go [hunk 2]
```diff
@@ -442,12 +430,14 @@ func TestServerNotify(t *testing.T) {
 	}
 	logger := logtest.NoOp(t)
 
-	conf, err := config.NewFromEnv()
-	if err != nil {
-		t.Fatal(err)
+	conf := config.Config{
+		BuildkiteToken: os.Getenv("BUILDKITE_WEBHOOK_TOKEN"),
+		SlackToken:     os.Getenv("SLACK_TOKEN"),
+		// GithubToken:    os.Getenv("GITHUB_TOKEN"),
```
Comment (burmudar) [inline dev/build-tracker/integration_test.go:436]:
```suggestion
```

File: dev/build-tracker/BUILD.bazel [hunk 3]
```diff
@@ -36,9 +47,58 @@ go_test(
...
+
+oci_push(
+    name = "candidate_push",
+    image = ":image",
+    repository = image_repository("build-tracker"),
+)
+
+msp_delivery(
+    name = "msp_deploy",
+    gcp_project = "build-tracker-prod-59bf",
```
Comment (burmudar) [inline dev/build-tracker/BUILD.bazel:101]: We probably need to adjust the dns record so that it points to this project + deployment

Comment (Strum355) [reply dev/build-tracker/BUILD.bazel:101]: I was gonna use `build-tracker.sgdev.org` so we could mirror the webhook traffic for testing live

**Pull request merged**